newrelic-alerts-configurator/src/main/java/com/ocadotechnology/newrelic/alertsconfigurator/configuration/PolicyConfiguration.java

```diff
@@ -23,35 +27,211 @@
  * </ul>
  */
 @Getter
-@Builder
 public class PolicyConfiguration {
     /**
      * Your policy name
      */
-    @NonNull
     private String policyName;
     /**
      * Rollup strategy options for your alerts policy
      */
-    @NonNull
     private IncidentPreference incidentPreference;
     /**
      * Collection of {@link Channel} configurations
      */
-    @Singular
-    private Collection<Channel> channels = Collections.emptyList();
+    private Collection<Channel> channels;
     /**
      * Collection of {@link Condition} configurations
      */
-    @Singular
-    private Collection<Condition> conditions = Collections.emptyList();
+    private Collection<Condition> conditions;
     /**
      * Collection of {@link ExternalServiceCondition} configurations
      */
-    @Singular
-    private Collection<ExternalServiceCondition> externalServiceConditions = Collections.emptyList();
+    private Collection<ExternalServiceCondition> externalServiceConditions;
+    /**
+     * Collection of {@link NrqlCondition} configurations
+     */
+    private Collection<NrqlCondition> nrqlConditions;
+
+    @ConstructorProperties(
+            {"policyName", "incidentPreference", "channels", "conditions", "externalServiceConditions", "nrqlConditions"})
+    PolicyConfiguration(String policyName,
```

Comment:
You can keep `@Builder` from lombok and only override a `build()` method

```
public static class PolicyConfigurationBuilder {
    public PolicyConfiguration build() {
        return new PolicyConfiguration(
                policyName,
                incidentPreference,
                normalizeList(this.channels),
                normalizeNullableList(this.conditions),
                normalizeNullableList(this.externalServiceConditions),
                normalizeNullableList(this.nrqlConditions));
    }

    private static <T> Collection<T> normalizeNullableList(List<T> list) {
        return list == null ? null : normalizeList(list);
    }

    private static <T> Collection<T> normalizeList(List<T> list) {
        switch (list == null ? 0 : list.size()) {
            case 0:
                return emptyList();
            case 1:
                return singletonList(list.get(0));
            default:
                return unmodifiableList(new ArrayList<>(list));
        }
    }
}
```

Reply:
Didn't know I can do that. Great idea!